dotNETStandard/HealthVaultProviderManagementPortal/HealthVaultProviderManagementPortal/Controllers/PatientController.cs

```diff
@@ -0,0 +1,159 @@
+using System.Web.Configuration;
+using HealthVaultProviderManagementPortal.Models.Patient;
+using Microsoft.HealthVault.Configuration;
+using Microsoft.HealthVault.Web.Attributes;
+using Microsoft.Rest;
+using Microsoft.Rest.Serialization;
+using Newtonsoft.Json;
+using NodaTime;
+using NodaTime.Serialization.JsonNet;
+using static HealthVaultProviderManagementPortal.Helpers.RestClientFactory;
+
+namespace HealthVaultProviderManagementPortal.Controllers
+{
+    /// <summary>
+    /// Controller for creating and sending invitations to remote patients
+    /// </summary>
+    [RequireSignIn]
+    public class PatientController : Controller
+    {
+        /// <summary>
+        /// Gets a the timeline for a patient
+        /// </summary>
+        public async Task<ActionResult> Index(Guid personId, Guid recordId, DateTime? startDate, DateTime? endDate)
+        {
+            if (startDate == null)
+            {
+                return View();
+            }
+
+            var timeline = await GetTimeline(personId, recordId, startDate.Value, endDate);
+
+            var timelineEntries = new List<TimelineEntryViewModel>();
+            foreach (var task in timeline.Tasks)
+            {
+                timelineEntries.AddRange(ConvertToTimelineEntryViewModels(task));
+            }
+
+            var groupedTimelineEntries = timelineEntries.OrderBy(t => t.LocalDateTime).ThenByDescending(t => t.ScheduleType).GroupBy(st => st.LocalDateTime.Date);
+
+            var model = new TimelineViewModel
+            {
+                StartDate = startDate,
+                EndDate = endDate,
+                TimelineEntries = groupedTimelineEntries
+            };
+
+            return View(model);
+        }
+
+        private async Task<TimelineResponse> GetTimeline(Guid personId, Guid recordId, DateTime startDate, DateTime? endDate)
+        {
+            var restHealthVaultUrl = WebConfigurationManager.AppSettings.Get("HV_RestHealthServiceUrl"); //TODO: use built in SDK function to retreive this
+
+            // Construct URL
+            var uriBuilder = new UriBuilder(restHealthVaultUrl)
+            {
+                Path = "Timeline"
+            };
+
+            var queryParameters = new List<string>
+            {
+                $"startDate={startDate.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture)}"
+            };
+
+            if (endDate != null)
+            {
+                queryParameters.Add($"endDate={endDate.Value.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture)}");
+            }
+
+            if (queryParameters.Count > 0)
+            {
+                uriBuilder.Query = string.Join("&", queryParameters);
+            }
+
+            // Create HTTP transport objects
+            var httpRequest = new HttpRequestMessage
+            {
+                Method = HttpMethod.Get,
+                RequestUri = uriBuilder.Uri,
+            };
+            httpRequest.Headers.Add("x-ms-version", "2.0-preview");
```

Comment:
I have a TODO to change to using SDK functionality to read the config (maybe I should be clearer). Otherwise this is as intended. It's preview functionality so the SDK won't expose it and we need to call it manually.